lib/core/test/unit/Cardano/Wallet/DB/TestWallet.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE StandaloneDeriving #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.DB.TestWallet where
+
+import Prelude
+
+import Control.DeepSeq
+    ( NFData )
+import GHC.Generics
+    ( Generic )
+import Test.QuickCheck
+    ( Arbitrary (..) )
+
+import qualified Data.ByteString.Char8 as B8
+import qualified Data.Text as T
+
+import Cardano.Wallet.Primitive.AddressDiscovery
+    ( IsOurs (..) )
+import Cardano.Wallet.Primitive.Model
+    ( Wallet, initWallet )
+import Cardano.Wallet.Primitive.Types
+    ( Hash (..)
+    , TxId (..)
+    , WalletDelegation (..)
+    , WalletMetadata (..)
+    , WalletName (..)
+    , WalletState (..)
+    )
+
+
+{-------------------------------------------------------------------------------
+                      Tests machinery, Arbitrary instances
+                               fixme: copy&pasted
+-------------------------------------------------------------------------------}
+
+data DummyTarget
+
+instance TxId DummyTarget where
+    txId = Hash . B8.pack . show
+
+newtype DummyState = DummyState Int
```

Comment:
`DummyState` probably should be like : 
```
newtype DummyStateSqlite = DummyStateSqlite (UTxO, PendingTx)
    deriving (Show, Eq, NFData)
```
Or even better to prepare test machine to take any state, ie. use state type parameter